src/client/systems/sysstarsystemrenderer.cpp

```diff
@@ -277,7 +280,40 @@ void conquerspace::client::systems::SysStarSystemRenderer::SeeEntity() {
     });
 }
 
+void conquerspace::client::systems::SysStarSystemRenderer::DrawEntityName(
+    glm::vec3 &object_pos, entt::entity ent_id) {
+
+    if (m_app.GetUniverse().all_of<conquerspace::common::components::Name>(
+        ent_id)) {
+    std::string &name =
+        m_app.GetUniverse()
+            .get<conquerspace::common::components::Name>(ent_id)
+            .name;
+    glm::vec3 pos =
+        glm::project(object_pos, camera_matrix, projection, viewport);
+
+    buffer_renderer.BeginDraw();
+    if (pos.z >= 1 || pos.z <= -1) {
+    } else {
+        m_app.DrawText(name, pos.x, pos.y);
+    }
+    buffer_renderer.EndDraw();
+    } else {
+        glm::vec3 pos =
+            glm::project(object_pos, camera_matrix, projection, viewport);
+        buffer_renderer.BeginDraw();
+        if (pos.z >= 1 || pos.z <= -1) {
+        } else {
+            m_app.DrawText(fmt::format("{}", ent_id), pos.x, pos.y);
+        }
+        buffer_renderer.EndDraw();
+    }
+}
```

Comment:
You can organize this a little better, it can look like this starting from line 295

```c++
buffer_renderer.BeginDraw();
if (m_app.GetUniverse().all_of<Name>(ent_id)) {
    std::string &name = m_app.GetUniverse().get<Name>(ent_id).name;
        glm::vec3 pos = glm::project(object_pos, camera_matrix, projection, viewport);

        if (!(pos.z >= 1 || pos.z <= -1)) {
            m_app.DrawText(name, pos.x, pos.y);
        }
    } else {
        glm::vec3 pos =
            glm::project(object_pos, camera_matrix, projection, viewport);
        if (!(pos.z >= 1 || pos.z <= -1)) {
            m_app.DrawText(fmt::format("{}", ent_id), pos.x, pos.y);
        }
    }
    buffer_renderer.EndDraw();
}
```

This keeps the start and end draw in a consistent place, and is more readable.